src/FileServiceManager/FileService/DAL/FileSystemConfiguration/FileSystemJsonConfiguration/FileSystemJsonConfigurator.cs

```diff
@@ -0,0 +1,229 @@
+                {
+                    JToken companiesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]");
+
+                    //if company node exists, add new file to files array
+                    if (companiesToken != null)
+                    {
+                        JArray filesArrayToken = (JArray)companiesToken.SelectToken($"$..files");
+                        filesArrayToken.Add(JToken.FromObject(fileMetadata));
+                    }
+                    else
+                    {
+                        //if company node does not exist, add company first, then files array to that company
+                        var compArray = (JArray)rootToken.SelectToken($"$.companies");
+
+                        CompanyMetadata company = new CompanyMetadata() { companyId = companyId, files = new List<FileMetadata>() { fileMetadata } };
+                        compArray.Add(JToken.FromObject(company));
+                    }
+
+                    using (fileWriter = File.CreateText(_configPath))
+                    {
+                        using (JsonTextWriter writer = new JsonTextWriter(fileWriter) { Formatting = Formatting.Indented })
+                        {
+                            await rootToken.WriteToAsync(writer);
+                        }
+                    }
+                }
+                else
+                {
+                    Log.Error("FileSystemConfig.json is not found");
+                }
+            }
+
+            catch (Exception ex)
+            {
+                Log.Error(ex.Message);
+            }
+            finally
+            {
+                fileReader?.Close();
+                fileWriter?.Close();
+            }
+        }
+
+        public async IAsyncEnumerable<FileMetadata> GetAllConfigurationsAsync(int companyId)
+        {
+            JToken rootToken = null;
+            StreamReader file = null;
+            JToken filesToken = null;
+            try
+            {
+                using (file = File.OpenText(_configPath))
+                {
+                    using (JsonTextReader reader = new JsonTextReader(file))
+                    {
+                        rootToken = await JToken.ReadFromAsync(reader);
+                    }
+                }
+
+                if (rootToken != null)
+                {
+                    filesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]")?.SelectToken($"$..files");
+                }
+                else
+                {
+                    Log.Error("FileSystemConfig.json is not found");
+                }
+            }
+
+            catch (Exception ex)
+            {
+                Log.Error(ex.Message);
+            }
+
+            finally
+            {
+                file?.Close();
+            }
+
+            if (filesToken == null)
+            {
+                Log.Error($"No file for company ID = {companyId}");
```

Comment:
in this case we need to pass to client not found or something like that

Reply:
+